ui/shared/components/panel/variants/Predictions.jsx

```diff
@@ -38,38 +43,72 @@ const Prediction = ({ field, annotation, dangerThreshold, warningThreshold }) =>
     }
   } else {
     color = SEVERITY_MAP[value]
-    value = value.replace('_', ' ')
+    if (value) {
+      value = `${value}`.replace('_', ' ')
+    }
+  }
+
+  if (!name) {
+    name = field.replace(/_/g, ' ').toUpperCase()
   }
 
-  return <div><Icon name="circle" size="small" color={color} />{field.replace('_', ' ').toUpperCase()} <PredictionValue>{value}</PredictionValue></div>
+  return (
+    <div>
+      <Icon name="circle" size="small" color={color} /> {name}
+      <PredictionValue> {value}</PredictionValue>
+    </div>)
 }
 
 Prediction.propTypes = {
   field: PropTypes.string.isRequired,
   annotation: PropTypes.object,
   dangerThreshold: PropTypes.number,
   warningThreshold: PropTypes.number,
+  name: PropTypes.string,
 }
 
+const MORE_PREDICTORS = [
+  'polyphen', 'sift', 'mut_taster', 'fathmm', 'metasvm', 'gerp_rs', 'phastcons100vert',
+]
 
-const Predictions = ({ annotation }) => {
-  return annotation ?
-    <div>
-      <Prediction field="polyphen" annotation={annotation} />
-      <Prediction field="sift" annotation={annotation} />
-      <Prediction field="mut_taster" annotation={annotation} />
-      <Prediction field="fathmm" annotation={annotation} />
-      <Prediction field="cadd_phred" annotation={annotation} dangerThreshold={20} warningThreshold={10} />
-      <Prediction field="dann_score" annotation={annotation} dangerThreshold={0.96} warningThreshold={0.93} />
-      <Prediction field="revel_score" annotation={annotation} dangerThreshold={0.75} warningThreshold={0.5} />
-      <Prediction field="eigen_phred" annotation={annotation} dangerThreshold={1} warningThreshold={2} />
-      <Prediction field="mpc_score" annotation={annotation} dangerThreshold={2} warningThreshold={1} />
-    </div>
-    : null
-}
+export default class Predictions extends React.Component {
+  static propTypes = {
+    annotation: PropTypes.object,
+  }
 
-Predictions.propTypes = {
-  annotation: PropTypes.object,
+  constructor(props) {
+    super(props)
+
+    this.state = { showMore: false }
+  }
+
+  render() {
+    const { annotation } = this.props
+
+    return annotation ?
+      <div>
+        <Prediction field="cadd_phred" annotation={annotation} dangerThreshold={20} warningThreshold={10} name="CADD" />
+        <Prediction field="dann_score" annotation={annotation} dangerThreshold={0.96} warningThreshold={0.93} name="DANN" />
+        <Prediction field="revel_score" annotation={annotation} dangerThreshold={0.75} warningThreshold={0.5} name="REVEL" />
+        <Prediction field="eigen_phred" annotation={annotation} dangerThreshold={1} warningThreshold={2}name="EIGEN" />
+        <Prediction field="mpc_score" annotation={annotation} dangerThreshold={2} warningThreshold={1} name="MPC" />
+        <Prediction field="primate_ai_score" annotation={annotation} dangerThreshold={0.7} warningThreshold={0.5} name="PMT AI" />
+        <Transition.Group animation="fade down" duration="500">
```

Comment:
I think the behavior here needs a bit of tweaking, because if only one of the "main" predictors is present and say 3 of the `MORE_PREDICTORS` then the better behavior would be to just show those 4 predictors with no toggle. So rather than having a hardcoded set of predictors that are always show and some that are always toggled, I think it would make more sense to have a list of predictors and then filter them by the presence of a value and then slice it so the top 6(? 8? whatever) are rendered and any remainders are toggleable

Reply:
can see some people preferring what you suggest, but not sure if everyone would since it's a little less predictable what gets shown where. Do you want to get feedback from people?

Reply:
If you feel strongly we can ask, but I feel pretty strongly about this one. If we were always showing all the predictors with some placeholder even if they were empty I would understand the desire to always have the same set shown vs hidden. However, given that we only show some of the predictors for each variant so there is no uniformity in whats shown anyways, I can't imagine a scenario where it is a better user experience to have to do an extra click